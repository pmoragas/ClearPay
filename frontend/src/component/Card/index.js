import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';
import { useForm } from 'react-hook-form';

const Card = ({wallet, onSubmit, errorMsg}) => {
    const { register, handleSubmit } = useForm();

    return (
        <div className={styles.container}>
            <div className={styles.title}>Transfer operation</div>
            <form onSubmit={handleSubmit(onSubmit)}>
                <div className={styles.subtitle}>Origin</div>
                <label className={styles.inputElem}>Wallet ID</label>
                <input
                    id="originWalletId"
                    label='Wallet ID'
                    name="originWalletId"
                    required
                    {...register('originWalletId')}
                    type="text"
                    className={styles.inputElem}
                    readOnly
                    value={wallet.id}
                />
                <div className={styles.dataPoint}>
                    <div className={styles.label}>Amount</div>
                    <div className={styles.data}>{wallet.amount}</div>
                </div>
                <div className={styles.subtitle}>Destination</div>
                <label className={styles.inputElem}>Wallet ID</label>
                <input
                    id="destinationWalletId"
                    label='Wallet ID'
                    name="destinationWalletId"
                    required
                    {...register('destinationWalletId')}
                    type="text"
                    className={styles.inputElem}
                />
                <label className={styles.inputElem}>Amount</label>
                <input
                    id="amount"
                    label='Amount'
                    name="amount"
                    required
                    {...register('amount')}
                    type="number"
                    className={styles.inputElem}
                />
                <div className={styles.buttonContainer}>
                    <div className={styles.error}>{errorMsg}</div>
                    <button id="submit" type="submit" className={styles.button}>
                        Submit
                    </button>
                </div>
            </form>
        </div>
    );
}

Card.propTypes = {
	wallet: PropTypes.shape({
        id: PropTypes.string,
        amount: PropTypes.number
    }),
    onSubmit: PropTypes.func,
    errorMsg: PropTypes.string
};

Card.default = {
	wallet: {},
    onSubmit: () => {},
	errorMsg: undefined
};

export default Card;
