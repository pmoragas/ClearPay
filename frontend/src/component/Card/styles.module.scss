@import '~styles/colors';

.container {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-family: sans-serif;
    width: 700px;
    height: 550px;
    border-radius: 15px;
    display: flex;
    background-color: $white;
    display: flex;
    flex-direction: column;

    .title {
        border-radius: 15px 15px 0 0;
        height:55px;
        width: 100%;
        background-color: $primary;
        font-weight: 600;
        font-size: 24px;
        text-align: center;
        line-height: 55px;
    }

    .subtitle {
        font-weight: 600;
        font-size: 20px;
        padding: 15px 0;
    }

    .dataPoint {
        display: flex;
        flex-direction: column;
        padding: 15px 0;

        .label {
            border-bottom: 1px solid black;
            font-weight: 600;
        }
    }

    border-top: 1px solid black;

        form {
            display: flex;
            flex-direction: column;
            padding: 0 30px;
            height: 100%;

            label {
                padding-top: 15px;
                font-weight: 600;
            }

            input {
                height: 40px;
                font-size: 16px;

            }

            .buttonContainer {
                display: flex;
                justify-content: center;
                align-items: center;
                flex-direction: column;
                height: 100%;

                .error {
                    color: red;
                }

                button {
                    width: 150px;
                    height: 40px;
                    font-weight: 600;
                    font-size: 16px;
                    border: unset;
                    border-radius: .75rem;
                    background-color: black;
                    color: #ffffff;

                    &:hover {
                        background-color: rgba($secondary, 0.5);
                    }
                }
            }
        }

    .destination {

    }
}
