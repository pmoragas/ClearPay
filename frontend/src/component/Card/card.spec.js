import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from './index';

describe('Button', () => {
	test('Render Card Text', () => {
		render(<Card wallet={{id: "abcdefgh", amount: 1000}} onSubmit={() => console.log('submitted')}/>);

		expect(screen.getByText('1000')).toBeInTheDocument();
		expect(screen.getByText('Submit')).toBeInTheDocument();
		expect(screen.getByDisplayValue('abcdefgh')).toBeInTheDocument();
	});

	test('snapshot is ok', () => {
		const { container } = render(<Card wallet={{id: "abcdefgh", amount: 1000}}/>);

		expect(container).toMatchSnapshot();
	});
});
