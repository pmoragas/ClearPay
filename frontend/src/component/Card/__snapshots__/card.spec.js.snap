// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot is ok 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      Transfer operation
    </div>
    <form>
      <div
        class="subtitle"
      >
        Origin
      </div>
      <label
        class="inputElem"
      >
        Wallet ID
      </label>
      <input
        class="inputElem"
        id="originWalletId"
        label="Wallet ID"
        name="originWalletId"
        readonly=""
        required=""
        type="text"
        value="abcdefgh"
      />
      <div
        class="dataPoint"
      >
        <div
          class="label"
        >
          Amount
        </div>
        <div
          class="data"
        >
          1000
        </div>
      </div>
      <div
        class="subtitle"
      >
        Destination
      </div>
      <label
        class="inputElem"
      >
        Wallet ID
      </label>
      <input
        class="inputElem"
        id="destinationWalletId"
        label="Wallet ID"
        name="destinationWalletId"
        required=""
        type="text"
      />
      <label
        class="inputElem"
      >
        Amount
      </label>
      <input
        class="inputElem"
        id="amount"
        label="Amount"
        name="amount"
        required=""
        type="number"
      />
      <div
        class="buttonContainer"
      >
        <div
          class="error"
        />
        <button
          class="button"
          id="submit"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
